bin/ruby-build

```diff
@@ -1119,18 +1133,45 @@ build_package_mac_openssl() {
   # gives precedence to the last -j option, so we can override that.
   package_option openssl make -j 1
 
-  # Use install_sw install_ssldirs instead of install to skip building docs which is slow.
-  # OpenSSL 1.1+ also needs install_ssldirs, 1.0 does not have that target.
-  if [[ "$1" == openssl-1.0.* ]]; then
-    MAKE_INSTALL_TARGET="install_sw" build_package_standard "$@"
+  if is_mac; then
+    # Use install_sw install_ssldirs instead of install to skip building docs which is slow.
+    # OpenSSL 1.1+ also needs install_ssldirs, 1.0 does not have that target.
+    if [[ "$1" == openssl-1.0.* ]]; then
+      MAKE_INSTALL_TARGET="install_sw" build_package_standard "$@"
+    else
+      MAKE_INSTALL_TARGET="install_sw install_ssldirs" build_package_standard "$@"
+    fi
   else
-    MAKE_INSTALL_TARGET="install_sw install_ssldirs" build_package_standard "$@"
+    build_package_standard "$@"
   fi
 
-  # Extract root certs from the system keychain in .pem format and rehash.
-  local pem_file="$OPENSSLDIR/cert.pem"
-  security find-certificate -a -p /Library/Keychains/System.keychain > "$pem_file"
-  security find-certificate -a -p /System/Library/Keychains/SystemRootCertificates.keychain >> "$pem_file"
+  if is_mac; then
+    # Extract root certs from the system keychain in .pem format and rehash.
+    local pem_file="$OPENSSLDIR/cert.pem"
+    security find-certificate -a -p /Library/Keychains/System.keychain > "$pem_file"
+    security find-certificate -a -p /System/Library/Keychains/SystemRootCertificates.keychain >> "$pem_file"
+  elif [ -e /etc/pki/tls/cert.pem ]; then # RedHat
+    # See https://github.com/rubygems/rubygems/issues/2415#issuecomment-509806259
+    rm -rf "$OPENSSLDIR/certs" "$pem_file"
+    ln -s /etc/pki/tls/certs "$OPENSSLDIR"
+    ln -s /etc/pki/tls/cert.pem "$pem_file"
+  elif [ -e /etc/ssl/certs/ca-certificates.crt ]; then # Debian
+    # See https://github.com/rubygems/rubygems/issues/2415#issuecomment-509806259
+    rm -rf "$OPENSSLDIR/certs" "$pem_file"
```

Comment:
Looks like `$pem_file` is only defined in the `if is_mac` clause, and thus it's empty here, causing the following `ln -s /etc/ssl/certs/ca-certificates.crt "$pem_file"` to fail.